Tests/SwiftParserTest/ExpressionTests.swift

```diff
@@ -2826,4 +2849,65 @@ final class StatementExpressionTests: ParserTestCase {
       experimentalFeatures: .typedThrows
     )
   }
+
+  func testArrayExprWithNoCommas() {
+    assertParse("[() ()]")
```

Comment:
This is parsed as 

```
(lldb) po el
▿ () ()
  ▿ raw : .layout(functionCallExpr byteLength=5 descendantCount=8)
    - 0 : nil
    ▿ 1 : .layout(tupleExpr byteLength=3 descendantCount=3)
      - 0 : nil
      - 1 : .parsedToken(leftParen wholeText="(" textRange=0..<1)
      - 2 : nil
      - 3 : .layout(labeledExprList byteLength=0 descendantCount=0)
      - 4 : nil
      - 5 : .parsedToken(rightParen wholeText=") " textRange=0..<1)
      - 6 : nil
    - 2 : nil
    - 3 : .parsedToken(leftParen wholeText="(" textRange=0..<1)
    - 4 : nil
    - 5 : .layout(labeledExprList byteLength=0 descendantCount=0)
    - 6 : nil
    - 7 : .parsedToken(rightParen wholeText=")" textRange=0..<1)
    - 8 : nil
    - 9 : nil
    - 10 : nil
    - 11 : .layout(multipleTrailingClosureElementList byteLength=0 descendantCount=0)
    - 12 : nil
```

It looks like it parses a `functionCallExpr` inside the and places the first `()` in `FunctionCallExprSyntax.unexpectedBeforeCalledExpression`

Reply:
I think it parses as follows, which seems fine to me. You are trying to call an expression that produces an empty tuple, which is fine as far as the parser is concerned. The type checker will eventually complain that `()`, aka `Void` is not callable.
```
swift-parser-cli print-tree -s '()()'
SourceFileSyntax
├─statements: CodeBlockItemListSyntax
│ ╰─[0]: CodeBlockItemSyntax
│   ╰─item: FunctionCallExprSyntax
│     ├─calledExpression: TupleExprSyntax
│     │ ├─leftParen: leftParen
│     │ ├─elements: LabeledExprListSyntax
│     │ ╰─rightParen: rightParen
│     ├─leftParen: leftParen
│     ├─arguments: LabeledExprListSyntax
│     ├─rightParen: rightParen
│     ╰─additionalTrailingClosures: MultipleTrailingClosureElementListSyntax
╰─endOfFileToken: endOfFile

```